frontend/src/pages/AuthCallbackPage.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useRef } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { message } from 'antd';
+import {KEYCLOAK_CLIENT_ID, KEYCLOAK_TOKEN_URL} from "../consts/apiConsts.ts";
+
+const AuthCallbackPage: React.FC = () => {
+    const [searchParams] = useSearchParams();
+    const navigate = useNavigate();
+    const hasHandledCallback = useRef(false);
+
+    useEffect(() => {
+        const handleAuthCallback = async () => {
+            if (hasHandledCallback.current) {
+                return;
+            }
+            hasHandledCallback.current = true;
+
+            const code = searchParams.get('code');
+            if (!code) {
+                message.error('No authorization code in URL');
+                navigate('/login');
+                return;
+            }
+
+            try {
+                console.log('Code:', code);
+                console.log('Redirect URI:', `${window.location.origin}/auth/callback`);
+
+                const response = await fetch(`${KEYCLOAK_TOKEN_URL}`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/x-www-form-urlencoded',
+                    },
+                    body: new URLSearchParams({
+                        grant_type: 'authorization_code',
+                        code: code,
+                        redirect_uri: `${window.location.origin}/auth/callback`,
+                        client_id: `${KEYCLOAK_CLIENT_ID}`,
+                    }),
+                });
+
+                if (!response.ok) {
+                    const errorText = await response.text();
+                    console.error('Keycloak response error:', errorText);
+                    throw new Error('Error while communication with Keycloak');
+                }
+
+                const data = await response.json();
+
+                if (!data.access_token || !data.refresh_token) {
+                    throw new Error('Response does not contains token!');
+                }
+
+                localStorage.setItem('access_token', data.access_token);
+                localStorage.setItem('refresh_token', data.refresh_token);
```

Comment:
Maybe you can create a const.
```typescript
const ACCESS_TOKEN_KEY = 'access_token';
const REFRESH_TOKEN_KEY = 'refresh_token';
```

![Ironic](https://github.com/user-attachments/assets/09a1065a-cf94-4cc5-9516-1ab3cdd2ea03)

Reply:
They occur exactly ***once*** in this file. If I import this in every file it will get so not good looking that I will have to had a bowl near me every time I do sth in frontend to have somewhere to vomit. In the case of Go, it makes sense, because I can import the whole module and get variables based on the module name, whereas in this abomination I have to import them directly. Is there any other way to do this? Is this the practice in this god forsaken language?

Reply:
Ask Vera or chatgpt. I never had to do this.

Reply:
Done